hamrs-pro: init at 2.33.0

ethancedwards8 opened the pull request:
Hamrs PRO is the next version of hamrs. It has both Linux and Darwin support.


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

ethancedwards8 committed:
hamrs-pro: init at 2.33.0

Hamrs PRO is the next version of hamrs. It has both Linux and Darwin
support.

Signed-off-by: Ethan Carter Edwards <ethan@ethancedwards.com>

ethancedwards8 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 390482`

Logs: https://github.com/ethancedwards8/nixpkgs-review-gha/actions/runs/13887037441

<details><summary>Download packages from cache:</summary><ul>
<li><details><summary><code>x86_64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-390482-x86_64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://ethancedwards8.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  ethancedwards8.cachix.org-1:YMasjqyFnDreRQ9GXmnPIshT3tYyFHE2lUiNhbyIxOc=
  ' \
  /nix/store/05z63ijn61lx0brldgq1crqhjbdg8mcv-hamrs-pro-2.33.0
```
</details></li>
<li><details><summary><code>aarch64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-390482-aarch64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://ethancedwards8.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  ethancedwards8.cachix.org-1:YMasjqyFnDreRQ9GXmnPIshT3tYyFHE2lUiNhbyIxOc=
  ' \
  /nix/store/zj0ri5h520jzk8mncz20qqs9w9jizw1w-hamrs-pro-2.33.0
```
</details></li>
<li><details><summary><code>x86_64-darwin</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-390482-x86_64-darwin \
  --option binary-caches 'https://cache.nixos.org/ https://ethancedwards8.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  ethancedwards8.cachix.org-1:YMasjqyFnDreRQ9GXmnPIshT3tYyFHE2lUiNhbyIxOc=
  ' \
  /nix/store/zv9cnv8wl0sl8awbadl2mnvn98y8qw80-hamrs-pro-2.33.0
```
</details></li>
<li><details><summary><code>aarch64-darwin</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-390482-aarch64-darwin \
  --option binary-caches 'https://cache.nixos.org/ https://ethancedwards8.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  ethancedwards8.cachix.org-1:YMasjqyFnDreRQ9GXmnPIshT3tYyFHE2lUiNhbyIxOc=
  ' \
  /nix/store/8p3yzyzc1h8zlv4w6vcl3zwffsxyd9rk-hamrs-pro-2.33.0
```
</details></li>
</ul></details>


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>hamrs-pro</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>hamrs-pro</li>
  </ul>
</details>


---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>hamrs-pro</li>
  </ul>
</details>


---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>hamrs-pro</li>
  </ul>
</details>

jhollowe commented on pkgs/by-name/ha/hamrs-pro/package.nix:
```diff
@@ -0,0 +1,31 @@
+Parks on the Air, Field Day, and more";
+    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
+    license = lib.licenses.unfree;
+    maintainers = with lib.maintainers; [ ethancedwards8 ];
```
feel free to add me as a maintainer too if you would like.

ethancedwards8 committed:
hamrs-pro: add jhollowe as a maintainer

Thanks!
Link: https://github.com/NixOS/nixpkgs/pull/390482#discussion_r1997891507

Signed-off-by: Ethan Carter Edwards <ethan@ethancedwards.com>

JohnRTitor commented on pkgs/by-name/ha/hamrs-pro/darwin.nix:
```diff
@@ -0,0 +1,40 @@
+stdenvNoCC.mkDerivation (finalAttrs: {
+  inherit pname version;
+
+  src =
+    if stdenvNoCC.hostPlatform.isAarch64 then
+      (fetchurl {
+        url = "https://hamrs-dist.s3.amazonaws.com/hamrs-pro-${finalAttrs.version}-mac-arm64.dmg";
+        hash = "sha256-H46z4V9lo+n/pZzna7KIiYxQBqTlZULitQrFEEMFDvo=";
+      })
+    else
+      (fetchurl {
+        url = "https://hamrs-dist.s3.amazonaws.com/hamrs-pro-${finalAttrs.version}-mac-x64.dmg";
+        hash = "sha256-KtrXF47AwVAUXYk1Wu2aKMTXENv7q9JBb86Oy+UHQYY=";
+      });
```
This is just bad, consider using something like `throwSystem`.

https://github.com/NixOS/nixpkgs/blob/babad4e1b472643b7694cca4017d5995f716dd15/pkgs/applications/version-management/bcompare/default.nix#L26-L40

ethancedwards8 replied:
It works, though. It is pretty easy to read too.

GaetanLepage replied:
I agree with @JohnRTitor it would be better to use `throwSystem`.

JohnRTitor commented on pkgs/by-name/ha/hamrs-pro/linux.nix:
```diff
@@ -0,0 +1,38 @@
+appimageTools.wrapType2 rec {
+  inherit pname version;
+
+  src =
+    if stdenvNoCC.hostPlatform.isAarch64 then
+      (fetchurl {
+        url = "https://hamrs-dist.s3.amazonaws.com/hamrs-pro-${version}-linux-arm64.AppImage";
+        hash = "sha256-YQPKxjaNXE1AgEspZRLI1OUFU71rAU8NBcS0Jv94MS8=";
+      })
+    else
+      (fetchurl {
+        url = "https://hamrs-dist.s3.amazonaws.com/hamrs-pro-${version}-linux-x86_64.AppImage";
+        hash = "sha256-FUwyyuXtWaHauZyvRvrH7KDC0du02eNR5TfKJyiKb9k=";
+      });
```
Same as above.

JohnRTitor reviewed (commented):
Also, both derivations can be in just one file:

Just like [google-chrome](https://github.com/NixOS/nixpkgs/blob/5e5402ecbcb27af32284d4a62553c019a3a49ea6/pkgs/by-name/go/google-chrome/package.nix#L316).

The .nix files are not nearly as big as to warrant splitting up like this.

An update script in the future, if feasible would be nice, it's just bad interface where the version in another file and hashes in another.

ethancedwards8 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 390482`

Logs: https://github.com/ethancedwards8/nixpkgs-review-gha/actions/runs/14206899543

<details><summary>Download packages from cache:</summary><ul>
<li><details><summary><code>x86_64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-390482-x86_64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://ethancedwards8.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  ethancedwards8.cachix.org-1:YMasjqyFnDreRQ9GXmnPIshT3tYyFHE2lUiNhbyIxOc=
  ' \
  /nix/store/hk8cglvr3pwqizl0237vlzkfgfkk7q5c-hamrs-pro-2.33.0
```
</details></li>
<li><details><summary><code>aarch64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-390482-aarch64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://ethancedwards8.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  ethancedwards8.cachix.org-1:YMasjqyFnDreRQ9GXmnPIshT3tYyFHE2lUiNhbyIxOc=
  ' \
  /nix/store/jv95a4416ygybybyr3kxin67b7w4p5lh-hamrs-pro-2.33.0
```
</details></li>
<li><details><summary><code>x86_64-darwin</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-390482-x86_64-darwin \
  --option binary-caches 'https://cache.nixos.org/ https://ethancedwards8.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  ethancedwards8.cachix.org-1:YMasjqyFnDreRQ9GXmnPIshT3tYyFHE2lUiNhbyIxOc=
  ' \
  /nix/store/5j0ac2vm7apq4c9svfrxfk7zpfw0vs8x-hamrs-pro-2.33.0
```
</details></li>
<li><details><summary><code>aarch64-darwin</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-390482-aarch64-darwin \
  --option binary-caches 'https://cache.nixos.org/ https://ethancedwards8.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  ethancedwards8.cachix.org-1:YMasjqyFnDreRQ9GXmnPIshT3tYyFHE2lUiNhbyIxOc=
  ' \
  /nix/store/5phwhhq1ylflh5hc2xgv2igzrbvhqrsy-hamrs-pro-2.33.0
```
</details></li>
</ul></details>


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>hamrs-pro</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>hamrs-pro</li>
  </ul>
</details>


---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>hamrs-pro</li>
  </ul>
</details>


---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>hamrs-pro</li>
  </ul>
</details>

ethancedwards8 commented:
Moved to throw system and one file.

GaetanLepage reviewed (approved):
tested locally on `x86_64-linux`.

ethancedwards8 commented on pkgs/by-name/ha/hamrs-pro/package.nix:
```diff
@@ -0,0 +1,99 @@
+
+  meta = {
+    homepage = "https://hamrs.app/";
+    description = "Simple, portable logger tailored for activities like
+Parks on the Air, Field Day, and more";
```
Oops. You're right. Nixfmt must have done that.

ethancedwards8 replied:
Fixed.
